lib/src/DynamicProto.ts

```diff
@@ -133,25 +143,27 @@ function _isObjectArrayOrFunctionPrototype(target:any)
  * @ignore
  */ 
 function _getObjProto(target:any) {
+    let newProto;
+
     if (target) {
         // This method doesn't existing in older browsers (e.g. IE8)
         if (_objGetPrototypeOf) {
             return _objGetPrototypeOf(target);
         }
 
-        // target[Constructor] May break if the constructor has been changed or removed
-        let newProto = target[DynProtoPolyProto] || target[str__Proto] || target[Prototype] || (target[Constructor] ? target[Constructor][Prototype] : null);
-        if(newProto) {
-            // Set the _dynProto so the hierarchy can lookup base classes correctly on IE8
-            if (!target[DynProtoPolyProto]) {
-                target[DynProtoPolyProto] = newProto;
-            }
+        let curProto = target[str__Proto] || target[Prototype] || (target[Constructor] ? target[Constructor][Prototype] : null);
```

Comment:
Note: This code is only reached for runtimes that don't support getPrototypeOf() https://caniuse.com/?search=getPrototypeOf